app/renderer/renderer.py

```diff
@@ -150,16 +158,22 @@ def _augment_questionnaire(self):
         # loops through the Schema and get errors and warnings in order
         # augments each item in the schema as required
         for group in self._schema.groups:
-            self._augment_item(group, errors, warnings)
-            for block in group.blocks:
-                self._augment_item(block, errors, warnings)
-                for section in block.sections:
-                    self._augment_item(section, errors, warnings)
-                    for question in section.questions:
-                        self._augment_item(question, errors, warnings)
-                        for response in question.responses:
-                            self._augment_response(response)
-                            self._augment_item(response, errors, warnings)
+            if self._current_group.id == group.id:
+                self._augment_item(group, errors, warnings)
+                for block in group.blocks:
+                    if self._current_block.id == block.id:
+                        self._augment_item(block, errors, warnings)
+                        for section in block.sections:
+                            self._augment_item(section, errors, warnings)
+                            for question in section.questions:
+                                self._augment_item(question, errors, warnings)
+                                for response in question.responses:
```

Comment:
Ah nevermind. I see the difference now, its the responses from the schema and the response store responses. They are different things (this is why hated that naming convention). Ok this makes sense now.